AppCenterDataStorage/AppCenterDataStorage/Internal/LocalStore/MSDBDocumentStore.m

```diff
@@ -37,18 +37,45 @@ - (instancetype)init {
     _expirationTimeColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSExpirationTimeColumnName]).unsignedIntegerValue;
     _downloadTimeColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSDownloadTimeColumnName]).unsignedIntegerValue;
     _operationTimeColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSOperationTimeColumnName]).unsignedIntegerValue;
-    _pendingOperationColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSPendingDownloadColumnName]).unsignedIntegerValue;
+    _pendingOperationColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSPendingOperationColumnName]).unsignedIntegerValue;
   }
   return self;
 }
 
 #pragma mark - Table Management
 
-// TODO work item created to track this implementation
-- (BOOL)createWithPartition:(NSString *)__unused partition
-                   document:(MSDocumentWrapper *)__unused document
-               writeOptions:(MSWriteOptions *)__unused writeOptions {
-  return YES;
+- (BOOL)upsertWithPartition:(NSString *)partition
+                  accountId:(NSString *_Nullable)accountId
+            documentWrapper:(MSDocumentWrapper *)documentWrapper
+                  operation:(NSString *_Nullable)operation
+                    options:(MSBaseOptions *)options {
+  NSDate *now = [NSDate date];
+  NSDate *expirationTime = [now dateByAddingTimeInterval:options.deviceTimeToLive];
+  NSString *tableName = [MSDBDocumentStore tableNameForPartition:partition accountId:accountId];
+  NSString *insertQuery =
+      [NSString stringWithFormat:@"REPLACE INTO '%@' ('%@', '%@', '%@', '%@', '%@', '%@', '%@', '%@') "
+                                 @"VALUES ('%@', '%@', '%@', '%@', '%@', '%@', '%@', '%@')",
+                                 tableName, kMSPartitionColumnName, kMSDocumentIdColumnName, kMSDocumentColumnName, kMSETagColumnName,
+                                 kMSExpirationTimeColumnName, kMSDownloadTimeColumnName, kMSOperationTimeColumnName,
+                                 kMSPendingOperationColumnName, partition, documentWrapper.documentId, documentWrapper.jsonValue,
+                                 documentWrapper.eTag, [MSUtility dateToISO8601:expirationTime],
+                                 [MSUtility dateToISO8601:documentWrapper.lastUpdatedDate], [MSUtility dateToISO8601:now], operation];
+  NSInteger result = [self.dbStorage executeNonSelectionQuery:insertQuery];
```

Comment:
Can use `int` instead of `NSInteger `, that's what executeNonSelectionQuery returns.
```suggestion
  int result = [self.dbStorage executeNonSelectionQuery:insertQuery];
```